### PR TITLE
feat: add top level interface

### PR DESCRIPTION
**Pull request by trym-b**
In order to get early feedback on the design of this project, only the
interface of the service is implemented by this commit.

This project will provide two features, continuously sending data to
digital storage and continuously verifying that the data sent to storage
has been properly stored.

The actual implementation details will be added and properly described
in later commits.

Each command both `send` and `verify` require that the user passes a
`kafka-endpoints` argument

Examples of the interfaces can be found below
```
$ hermetic --help
hermetic - sends and verifies data for digital storage

Usage:
  hermetic [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  send        Continuously sends data to digital storage
  verify      Continuously verifies uploaded data responses

Flags:
  -h, --help                      help for hermetic
      --kafka-endpoints strings   list of kafka endpoints

Use "hermetic [command] --help" for more information about a command.
```

```
$ hermetic send --help
Continuously sends data to digital storage

Usage:
  hermetic send [flags]

Flags:
  -h, --help                          help for send
      --stage-artifacts-root string   path to the root directory with the following content structure:
                                      /stage-artifacts-root
                                      ├── /kommuner_2023-20230611002729-0035-veidemann-contentwriter-568c6f8545-frvcm
                                      │   ├── /kommuner_2023-20230611002729-0035-veidemann-contentwriter-568c6f8545-frvcm.warc.gz
                                      │   └── /checksum_transferred.md5
                                      └── /kommuner_2023-20230611002730-0036-veidemann-contentwriter-568c6f8545-frvcm
                                          ├── /kommuner_2023-20230611002730-0036-veidemann-contentwriter-568c6f8545-frvcm.warc.gz
                                          └── /checksum_transferred.md5
                                      ... etc
      --transfer-topic string         name of transfer-topic

Global Flags:
      --kafka-endpoints strings   list of kafka endpoints
```

```
$ hermetic verify --help
Continuously verifies uploaded data responses

Usage:
  hermetic verify [flags]

Flags:
  -h, --help                  help for verify
      --reject-topic string   name of reject-topic

Global Flags:
      --kafka-endpoints strings   list of kafka endpoints
```

When some of the arguments are missing, the output will look like this
```
$ hermetic --kafka-endpoints 123,456 send
Error: required flag(s) "transfer-topic" not set
Usage:
  hermetic send [flags]

Flags:
  -h, --help                          help for send
      --stage-artifacts-root string   path to the root directory with the following content structure:
                                      /stage-artifacts-root
                                      ├── /kommuner_2023-20230611002729-0035-veidemann-contentwriter-568c6f8545-frvcm
                                      │   ├── /kommuner_2023-20230611002729-0035-veidemann-contentwriter-568c6f8545-frvcm.warc.gz
                                      │   └── /checksum_transferred.md5
                                      └── /kommuner_2023-20230611002730-0036-veidemann-contentwriter-568c6f8545-frvcm
                                          ├── /kommuner_2023-20230611002730-0036-veidemann-contentwriter-568c6f8545-frvcm.warc.gz
                                          └── /checksum_transferred.md5
                                      ... etc
      --transfer-topic string         name of transfer-topic

Global Flags:
      --kafka-endpoints strings   list of kafka endpoints

failed to execute command, got error: 'required flag(s) "transfer-topic" not set'
exit status 1
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,14 +9,12 @@ RUN go mod download
 
 COPY . .
 
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o /ping-server
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o /hermetic
 
 FROM golang:1.20 AS run-stage
```
Comment (maeb) [inline Dockerfile:14]:
```suggestion
FROM gcr.io/distroless/base-debian11
```

We prefer distroless images.

Don't see any reason to use `AS run-stage` as it is implicit and not referenced anywhere.

**Review by maeb — COMMENTED** [review]
Look good, see comment.

File: Dockerfile (hunk 1 above)
Comment (trym-b) [reply Dockerfile:14]: As this is a change that is strictly unreleated to the code this PR changes, I will change the image we use in a separate PR. Sounds good?

**Comment by trym-b** [comment]
Added a fixup, I will rewrite the commit message to also mention the `confirm` topic messages when approved.

File: Dockerfile (hunk 1 above)
Comment (trym-b) [reply Dockerfile:14]: Solved by https://github.com/nlnwa/hermetic/pull/4

**Comment by trym-b** [comment]
Added a commit that added a missing argument for transfer `--root-path`, the commit message will be changed later

**Comment by trym-b** [comment]
Ready for next round.

**Review by johnerikhalse — APPROVED** [review]
LGTM

**Pull request merged**